homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""Sensor to read Proxmox VE data."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import DATA_GIBIBYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import COORDINATORS, DOMAIN, PROXMOX_CLIENTS, ProxmoxEntity
+
+BYTE_TO_GIBIBYTE = 1.074e9
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None,
+) -> None:
+    """Set up sensors."""
+
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    for host_config in discovery_info["config"][DOMAIN]:
+        host_name = host_config["host"]
+        host_name_coordinators = hass.data[DOMAIN][COORDINATORS][host_name]
+
+        if hass.data[PROXMOX_CLIENTS][host_name] is None:
+            continue
+
+        for node_config in host_config["nodes"]:
+            node_name = node_config["node"]
+
+            for dev_id in node_config["vms"] + node_config["containers"]:
+                coordinator = host_name_coordinators[node_name][dev_id]
+
+                # unfound case
+                if (coordinator_data := coordinator.data) is None:
+                    continue
+
+                name = coordinator_data["name"]
+
+                sensors += [
+                    ProxmoxSensor(
+                        coordinator=coordinator,
+                        unique_id=f"proxmox_{node_name}_{dev_id}_mem_gib",
+                        name=f"{node_name}_{name}_memory",
+                        icon="",
+                        host_name=host_name,
+                        node_name=node_name,
+                        vm_id=dev_id,
+                        native_lambda=lambda data: round(
+                            int(data["mem"]) / BYTE_TO_GIBIBYTE, 2
+                        ),
+                        unit_of_measurement=DATA_GIBIBYTES,
+                    )
+                ]
```

Comment:
Since you are planning to add more sensors, I would say this is a perfect fit for using EntityDescription

Reply:
I think you are right, but incorporating EntityDescription right now would require some re-working of binary_sensor, which is a bit out of the scope of this story. Do you think it would be acceptable to pospone this suggestion for a future PR (which I would focus on right after this one is complete) that would involve cleaning a lot of this up?

I wrote most of this integration a few years ago and I'm really wanting to fix a lot of it now because it wasn't done the best lol

Reply:
You can add EntityDescription to the sensor platform without the need to implement it on the BinarySensor platform.
Since you are adding a new platform, I think it makes sense to implement it on this new platform.